### PR TITLE
Update kubectl drain help message

### PR DESCRIPTION
**Pull request by mengqiy**
Update `kubectl drain` help messages according to kubernetes/kubernetes.github.io#1768

cc: @erictune @pwittrock

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37321)

### CODE REVIEW

File: pkg/kubectl/cmd/drain.go [hunk 1]
```diff
@@ -140,6 +140,9 @@ var (
 		Drain node in preparation for maintenance.
 
 		The given node will be marked unschedulable to prevent new pods from arriving.
+		'drain' evicts the pods if the APIServer supports eviciton
+		(http://kubernetes.io/docs/admin/disruptions/). Otherwise, it will use normal DELETE
+		to delete the pods.
 		The 'drain' deletes all pods except mirror pods (which cannot be deleted through
```
Comment (janetkuo) [inline pkg/kubectl/cmd/drain.go]: Above it says either evict or delete pods, but here it says deletes pods -- would that be confusing?

File: pkg/kubectl/cmd/drain.go [hunk 2]
```diff
@@ -149,6 +152,9 @@ var (
 		ReplicaSet, DaemonSet or Job, then drain will not delete any pods unless you
```
Comment (janetkuo) [inline pkg/kubectl/cmd/drain.go]: Should we mention StatefulSet here, in the example, and in the flag description?

Comment (janetkuo) [inline pkg/kubectl/cmd/drain.go]: Also in the warning / error messages

**Comment by ymqytw** [comment]
@janetkuo Updated. PTAL

**Comment by erictune** [comment]
LGTM

**Comment by ymqytw** [comment]
@k8s-bot verify test this

**Comment by erictune** [comment]
This needs to go to release-1.5 as well.

**Comment by dims** [comment]
@k8s-bot test this

**Review by janetkuo — APPROVED** [review]

**Comment by ymqytw** [comment]
> @k8s-bot test this

@dims The bot seems to be not responding to the command.

**Comment by ymqytw** [comment]
@k8s-bot test this

**Comment by dims** [comment]
@k8s-bot test this issue: #IGNORE

**Comment by ymqytw** [comment]
@k8s-bot test this

**Pull request merged**